### PR TITLE
Updated Navbar and Home UI

### PR DESCRIPTION
**Pull request by aryanchari06**
Updated UI
Kept an option for optional component rendering based on user role. (We haven't added User role in sessions so commented it)

### CODE REVIEW

File: src/app/layout.tsx [hunk 1]
```diff
@@ -28,11 +31,17 @@ export default function RootLayout({
       <body
         className={`${geistSans.variable} ${geistMono.variable} antialiased`}
       >
-        <main className={`${geistSans.className} h-screen w-screen flex flex-col justify-center items-center bg-gradient-to-tr from-indigo-900 to-purple-600`}>
```
Comment (yashranjan1) [inline src/app/layout.tsx:31]: why did you remove the flex attributes and the screen classes, theyre meant for positioning

File: src/components/Navbar.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  return (
+    <nav className="p-4 md:p-6 shadow-md bg-gray-900 text-white">
+      <div className="container mx-auto flex flex-col md:flex-row justify-between items-center">
+        <a href="/" className="text-xl font-bold mb-4 md:mb-0">
+          AutoResQ
+        </a>
+        {session ? (
+          <>
+            <span className="mr-4">Welcome, {user.username || user.email}</span>
```
Comment (yashranjan1) [inline src/components/Navbar.tsx]: wouldnt a user avatar that can be clicked to open a dropdown menu like on youtube be more appropriate here? have the signout on the dropdown in there?

File: src/context/AuthProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,6 @@
+"use client";
+
+import { SessionProvider } from "next-auth/react";
+export default function App({ children }: { children: React.ReactNode }) {
```
Comment (yashranjan1) [inline src/context/AuthProvider.tsx]: whats the point of this element, could you not wrap all the elements in layout with sessionprovider

File: src/app/layout.tsx (hunk 1 above)
Comment (aryanchari06) [reply src/app/layout.tsx:31]: i think its better to give the css positioning to each page's layout, so that if we want to change the design on some page it will be easier to do so.

File: src/components/Navbar.tsx (hunk 2 above)
Comment (aryanchari06) [reply src/components/Navbar.tsx]: yes we can do that

File: src/context/AuthProvider.tsx (hunk 3 above)
Comment (aryanchari06) [reply src/context/AuthProvider.tsx]: I had doubts regarding that, so i checked on the tutorial how it was done. I'll update it

File: src/app/layout.tsx (hunk 1 above)
Comment (yashranjan1) [reply src/app/layout.tsx:31]: You cant access the elements outside children from inside children so you would not be able to set the position of anything that is outside children from inside children, the display attributes allow for a positioning guideline for all pages so we dont have to bother with every page. 

Even if we were somehow able to influence the positioning from inside children to the outside, we would have to manually setup the positioning for every page, TONS of repeated code

If your aim to change what display type we use tho then thats a diff discussion, either way, we need a display attr and some height and width properties

Comment (aryanchari06) [reply src/app/layout.tsx:31]: What I meant was, we can give each page, the positioning. Ik it will be repetitive but in case we want to adjust the position of a particular page, it will be easier to do so. 
If that can be countered somehow, then I dont mind using the css in main itself.

Comment (yashranjan1) [reply src/app/layout.tsx:31]: oh, each page will have its own positioning, this is just meant to allot a certain amount of space to navbar children and footer in the layout, now that you say that tho i think it would be better to use min-h-screen over h-screen

**Comment by yashranjan1** [comment]
lgtm, merging
